webpack/scenes/Subscriptions/Manifest/CdnConfigurationTab/UpstreamServerTypeForm.js

```diff
@@ -48,37 +46,51 @@ const CdnConfigurationForm = (props) => {
   const [lifecycleEnvironmentLabel, setLifecycleEnvironmentLabel] =
     useState(cdnConfiguration.upstream_lifecycle_environment_label);
 
+  const [updateEnabled, setUpdateEnabled] =
+    useState(type !== cdnConfiguration.type);
+
+  const firstUpdate = useRef(true);
```

Comment:
I will have to do something like
```javascript
  const updateDisabled = () => {
    return (
      url === cdnConfiguration.url &&
      username === cdnConfiguration.username &&
      organizationLabel === cdnConfiguration.upstream_organization_label &&
      lifecycleEnvironmentLabel === cdnConfiguration.upstream_lifecycle_environment_label &&
      contentViewLabel === cdnConfiguration.upstream_content_view_label &&
      sslCaCredentialId === cdnConfiguration.ssl_ca_credential_id &&
      (password === null || password === cdnConfiguration.password)
    );
  }
```
and remove useRef and useEffect

Reply:
let me know if that is preferable

Reply:
I think the useEffect is great.. Would it work if we kept the useEffect but just used state for `firstUpdate`?

Reply:
interesting this seems to work
``` javascript
  const [updateEnabled, setUpdateEnabled] = useState(null);
  useEffect(() => {
    if (updateEnabled === null) {
      setUpdateEnabled(false);
      return;
    }
     setUpdateEnabled(true);
  }, [url, username, password, organizationLabel,
    contentViewLabel, lifecycleEnvironmentLabel,
    sslCaCredentialId]);

```

Reply:
Reverted back to first update because react gave me this warning
`  57:6  warning  React Hook useEffect has a missing dependency: 'updateEnabled'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
`